tree/ntuple/v7/src/RField.cxx

```diff
@@ -477,18 +477,19 @@ void ROOT::Experimental::Detail::RFieldBase::ConnectPageSink(RPageSink &pageSink
 
    /// Fix-up default encoding: if the ntuple is uncompressed, the default encoding should be non-split
    if ((pageSink.GetWriteOptions().GetCompression() == 0) && HasDefaultColumnRepresentative()) {
-      const auto &rep = GetColumnRepresentative();
-      if (rep == ColumnRepresentation_t({EColumnType::kSplitReal64})) {
-         SetColumnRepresentative({EColumnType::kReal64});
-      } else if (rep == ColumnRepresentation_t({EColumnType::kSplitReal32})) {
-         SetColumnRepresentative({EColumnType::kReal32});
-      } else if (rep == ColumnRepresentation_t({EColumnType::kSplitInt64})) {
-         SetColumnRepresentative({EColumnType::kInt64});
-      } else if (rep == ColumnRepresentation_t({EColumnType::kSplitInt32})) {
-         SetColumnRepresentative({EColumnType::kInt32});
-      } else if (rep == ColumnRepresentation_t({EColumnType::kSplitInt16})) {
-         SetColumnRepresentative({EColumnType::kInt16});
+      ColumnRepresentation_t rep = GetColumnRepresentative();
+      for (auto &colType : rep) {
+         switch (colType) {
+         case EColumnType::kSplitIndex32: colType = EColumnType::kIndex32; break;
+         case EColumnType::kSplitReal64: colType = EColumnType::kReal64; break;
+         case EColumnType::kSplitReal32: colType = EColumnType::kReal32; break;
+         case EColumnType::kSplitInt64: colType = EColumnType::kInt64; break;
+         case EColumnType::kSplitInt32: colType = EColumnType::kInt32; break;
+         case EColumnType::kSplitInt16: colType = EColumnType::kInt16; break;
+         default: break;
+         }
```

Comment:
Given that the loop breaks at the first iteration in any case, shouldn't we just take `rep[0]`?

Reply:
We may, but it would not be much simpler because we may not have a column at all (for struct fields).